components/script/script_thread.rs

```diff
@@ -1635,7 +1635,13 @@ impl ScriptThread {
     fn handle_frame_load_event(&self, parent_id: PipelineId, frame_id: FrameId, child_id: PipelineId) {
         let iframe = self.documents.borrow().find_iframe(parent_id, frame_id);
         match iframe {
-            Some(iframe) => iframe.iframe_load_event_steps(child_id),
+            Some(iframe) => {
+                // If the iframe has no "src" attribute, a load event will have already been fired
+                // when binding the iframe to the tree.
+                if iframe.upcast::<Element>().has_attribute(&local_name!("src")) {
```

Comment:
I have been thinking about this and testing a bit more... I don't know if this will be a sufficient solution. Consider the following case:

 * You have the `iframe` with no `src` 
 * The load event steps for the initial `about:blank` are fired async as per the spec
 * In a script you set the `src` of the `iframe` to some URL
 * The load event steps are now processed and a `load` event is fired for the initial `about:blank`

A possible fix is to cancel the load event steps runnable when the src is set.
I have been working on `about:blank` fixes (#14764) and here is a commit that does this https://github.com/servo/servo/pull/14764/commits/46220d409e75ddedefa3585115c3ea89a1e74b25
I would be ok if you wanted to implement something like that here since I don't know how soon I will have enough time to finish up the rest of that PR